content/articles/serialization-in-csharp/index.md

```diff
@@ -0,0 +1,169 @@
+
+2. **XML serialization**
+
+The public members of a class instance may be serialized to an XML stream. Serializing XML takes longer than serializing binary files. XML Serialization is used to enable cross-platform compatibility. Serializing XML is textual. XML files are easy to read and edit. XmlAttribute provides a serialization attribute that accepts XML.
+
+
+The XmlSerializer class uses the XML Serialization standard to serialize an object. Here's how to use XmlSerializer:
+
+```C#
+XmlSerializer Serializer = new XmlSerializer(typeof(Push));
+using (TextWriter Writer = new StreamWriter(@"c:\xmlseeralizer.xml"))
+{
+xmlSerializer.Serialize(Writer, pushObject);
+}
+```
+
+3. **SOAP serialization**
+Simple Object Access Protocol (SOAP) It is an XML-based protocol for data transmission between computers and offers the transport mechanism for web services. SOAP is language and platform agnost.
+SOAP serialization is used when we need to move items that are made up of incompatible designs, from one application to another.
+
+Serializable tells the.NET runtime that this class's instances can be serialized. This property is required for class-level serialization. In the Main() function of our Program.cs class, we'll utilize the SoapFormatter to serialize an instance of our Sample object into a Soap XML stream.
+
+Example program
+
+```c#
+// For soap serialization to occur, one must reference the soap serialization using ``using System.Runtime.Serialization.Formatters.Soap;``
+using system;
+using System.Text;
+using System.IO;
+using System.Runtime.Serialization;
+using System.collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Runtime.Serialization.Formatters.Soap;
+
+namespace soapSerializationSample
+{
+    class demo
+    {
+        static void Main(string[]args)
+        {
+            Sample sample = new Sample();
+            sample.Name = "erick Kiragu"; //Instance of our sample class
+            sample.Value =34;
+            FileStream filestream =new Filestream //This code is used to accept our output
+            (@"d:\sem\serialisation.dat", FileMode.Create);// Serialization of the created object
+            SoapFormatter formatter =new  SoapFormatter();
+            formatter.Serializa(fileStream, sample);
+        }
+    }
+}
+
+
+```
+
+From the above code, we created an instance of our sample class and called it erick nyaga, and then we serialized it using the value 34. After that we create a file stream to accept our output using the code `FileStream filestream =new Filestream(@"d:\sem\serialisation.dat", FileMode.Create);`. The line of the code after creating a file stream is used to serialize the object created. The final line of code serializes the objects to .data file format. The lines `using System.Runtime.Serialization;` and `using System.Runtime.Serialization.Formatters.Soap;` are the references for the code to do soap serialization.

```

Comment:
Quote the actual line(s) of code you're referring to and avoid using different tenses in the same paragraph/sentence.